src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -97,11 +103,11 @@ void IDkmClrResultProvider.GetItems(DkmEvaluationResultEnumContext enumContext,
                     return;
                 }
 
-                completionRoutine(GetItems(enumContext.InspectionContext, dataItem.EvalResultDataItem, startIndex, count));
+                completionRoutine(GetItems(dataItem.EvalResultDataItem, workList, startIndex, count, enumContext.InspectionContext));
             }
-            catch (Exception e) when (ExpressionEvaluatorFatalError.CrashIfFailFastEnabled(e))
+            catch (Exception e) when (ExpressionEvaluatorFatalError.ReportNonFatalException(e, DkmComponentManager.ReportCurrentNonFatalException))
             {
-                throw ExceptionUtilities.Unreachable;
+                completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e));
```

Comment:
For the methods that have a completion routine, I believe we have to catch all Exceptions and call the completion.  I cannot explain why these are different than the others, but I do know that throwing any Exception (and not calling the completion routine) will result in a hang.